### PR TITLE
refactor: `completeMultipartUpload` to TypeScript

### PR DESCRIPTION
**Pull request by trim21**

### CODE REVIEW

**Comment by trim21** [comment]
@prakashsvmx /ping

**Review by prakashsvmx — APPROVED** [review]

**Comment by trim21** [comment]
merge conflict resolved

**Comment by trim21** [comment]
@kaankabalak

File: src/internal/client.ts [hunk 1]
```diff
@@ -1139,6 +1139,73 @@ export class TypedClient {
...
+    }
+    if (!isValidObjectName(objectName)) {
+      throw new errors.InvalidObjectNameError(`Invalid object name: ${objectName}`)
+    }
+    if (!isString(uploadId)) {
+      throw new TypeError('uploadId should be of type "string"')
+    }
+    if (!isObject(etags)) {
+      throw new TypeError('etags should be of type "Array"')
+    }
```
Comment (kaankabalak) [inline src/internal/client.ts:1165]: As we now have types, do we need the check here?

Comment (kaankabalak) [inline src/internal/client.ts:1165]: As we now have types, do we still need these checks?

Comment (trim21) [reply src/internal/client.ts:1165]: there are some developers who doesn't use typescript

Comment (trim21) [reply src/internal/client.ts:1165]: I'm okay to remove these checks either

Comment (prakashsvmx) [reply src/internal/client.ts:1165]: we can leave it for now..

**Review by kaankabalak — APPROVED** [review]
LGTM

**Pull request merged**